azext_iot/operations/pnp.py

```diff
@@ -0,0 +1,243 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Unpublished works.
+# --------------------------------------------------------------------------------------------
+
+import json
+from os.path import exists
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_iot.common.shared import SdkType, PnPModelType
+from azext_iot.common._azure import get_iot_pnp_connection_string
+from azext_iot.pnp_sdk.models import SearchOptions
+from azext_iot._factory import _bind_sdk
+from azext_iot._constants import PNP_API_VERSION, PNP_ENDPOINT
+from azext_iot.common.utility import (unpack_pnp_http_error,
+                                      get_sas_token,
+                                      shell_safe_json_parse)
+from azure.cli.core.util import read_file_content
+
+logger = get_logger(__name__)
+
+
+def iot_pnp_interface_publish(cmd, interface, repo_endpoint=PNP_ENDPOINT, repo_id=None, login=None):
+    _validate_repository(repo_id, login)
+    model_list = _iot_pnp_model_list(cmd, repo_endpoint, repo_id, interface, PnPModelType.interface,
+                                     -1, login=login)
+    model_list_count = len(model_list)
+    if model_list_count > 0 and model_list[0].urn_id == interface:
```

Comment:
`if model_list and model_list[0].urn_id == interface`, then remove `model_list_count`

Reply:
Done